src/main/java/com/stripe/net/RequestTelemetry.java

```diff
@@ -26,12 +26,18 @@ class RequestTelemetry {
    * metrics, or if telemetry is disabled, then the returned {@code Optional} is empty.
    *
    * @param headers the request headers
+   * @deprecated Use {$ling {@link #pollPayload()} instead.
    */
+  @Deprecated
   public Optional<String> getHeaderValue(HttpHeaders headers) {
```

Comment:
Any reason we can't reuse the original method?

Reply:
Nothing *prohibitive*, but the original method is awkward and makes the calling code awkward.

First, I think checking for duplicate telemetry headers is odd in the first place since there's really only one place it gets set.

Second, if we're going to check for duplicate headers, the same class should be responsible for setting the header and checking for duplicates. It's very awkward that `getHeaderValue` returns an `Optional<String>` i.e. it leaves the responsibility for actually putting the value onto the `HEADER_NAME` header inside an HttpHeaders object to the caller -- but accepts a `HttpHeaders` object and checks `HEADER_NAME` there directly.

Reply:
Merging, but we can continue discussing and if I'm totally off-base we can bring it back in a follow-up.